src/Workspaces/Core/Portable/Editing/SyntaxGenerator.cs

```diff
@@ -1675,6 +1675,70 @@ public SyntaxNode TypeExpression(ITypeSymbol typeSymbol, bool addImport)
         public abstract SyntaxNode NullableTypeExpression(SyntaxNode type);
 
         /// <summary>
+        /// Creates an expression that denotes a tuple type.
+        /// </summary>
+        public abstract SyntaxNode TupleTypeExpression(IEnumerable<SyntaxNode> elements);
+
+        protected static void ValidateTupleElements(IEnumerable<SyntaxNode> elements)
```

Comment:
I think this doesn't need to be part of public API. You could use "method template design pattern" or just make `TupleTypeExpression` virtual and call `base` in the overrides.